TwitchClients/auth/BrowserLogin.cs

```diff
@@ -0,0 +1,196 @@
+                byte[] buffer = System.Text.Encoding.UTF8.GetBytes(responseString);
+
+                // send the output to the client browser
+                httpResponse.ContentLength64 = buffer.Length;
+                System.IO.Stream output = httpResponse.OutputStream;
+                output.Write(buffer, 0, buffer.Length);
+                output.Close();
+
+                httpListener.BeginGetContext(new AsyncCallback(IncomingHttpRequest), httpListener);
+            }else
+            {
+                Console.WriteLine("Twitch Interaction Oauth:: Got User Redirect, informing them they can close browser");
+                responseString = "<html><body><b>DONE!</b><br>(Please close this tab/window and restart SUBNAUTICA)</body></html>";
+
+                byte[] buffer = System.Text.Encoding.UTF8.GetBytes(responseString);
+
+                // send the output to the client browser
+                httpResponse.ContentLength64 = buffer.Length;
+                System.IO.Stream output = httpResponse.OutputStream;
+                output.Write(buffer, 0, buffer.Length);
+                output.Close();
+
+                Console.WriteLine("Twitch Interaction Oauth:: Done");
+                // the HTTP listener has served it's purpose, shut it down
+                httpListener.Stop();
+                _config.ClientId = twitchClientId;
+                _config.UsernameToken = _code;
+                GetUserId();
+            }
+        }
+
+        /// <summary>
+        /// Makes the API call to exchange the received code for the actual auth token
+        /// </summary>
+        /// <param name="code">The code parameter received in the callback HTTP reuqest</param>
+        internal void GetUserId()
+        {
+
+            Console.WriteLine("Twitch Interaction Oauth:: Getting User Id");
+
+            // check that we got a code value and the state value matches our remembered one
+            if ((_code.Length > 0) && (_state == _twitchAuthStateVerify))
+            {
+                string apiUrl;
+
+                // construct full URL for API call
+                apiUrl = "https://api.twitch.tv/helix/users";
+
+                HttpWebRequest request = (HttpWebRequest)WebRequest.Create(apiUrl);
+                request.Method = "GET";
+                request.Headers.Add("Authorization", "Bearer " + _code);
+                request.Headers.Add("client-id", twitchClientId);
+                try
+                {
+                    HttpWebResponse response = (HttpWebResponse)request.GetResponse();
+                    StreamReader reader = new StreamReader(response.GetResponseStream());
+                    string jsonResponse = reader.ReadToEnd();
+
+
+
+                    var objectResponse = JObject.Parse(jsonResponse);
+                    _config.UsernameId = (string)objectResponse["data"][0]["id"];
+                    _config.Username = (string)objectResponse["data"][0]["login"];
```

Comment:
Pretty sure that you can add:
`_config.Save(); `

here and it is going to ensure that these values are written out to the config.json file properly. Because we are using a custom handler for the config event, I'm pretty sure they are not persisted automatically.